### PR TITLE
[Securitycenter] Add tests

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by dvbb** [comment]
Linked issue:
[[SecurityCenter] The definition of SWAGGER is inconsistent with the actual return of Rest API #21260](https://github.com/Azure/azure-rest-api-specs/issues/21260)
[[Security]ComplianceResults_List Operation will get an incorrect response #21144](https://github.com/Azure/azure-rest-api-specs/issues/21144)

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/tests/Scenario/ConnectorSettingTests.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        [SetUp]
+        public void TestSetUp()
+        {
+        }
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        [RecordedTest]
+        [Ignore("System.InvalidOperationException : The requested operation requires an element of type 'Object', but the target element has type 'Array'.")]
```
Comment (fengzhou-msft) [inline sdk/securitycenter/Azure.ResourceManager.SecurityCenter/tests/Scenario/ConnectorSettingTests.cs:33]: Is this also swagger or service issue? Can you create a github issue if yes.

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/tests/Scenario/ResourceGroupLocationAlertTests.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        //    _resourceGroup = await CreateResourceGroup();
+        //    _resourceGroupLocationAlertCollection = _resourceGroup.GetResourceGroupSecurityAlerts();.GetResourceGroupLocationAlerts(DefaultLocation);
+        //}
+
+        //[RecordedTest]
+        //public async Task GetAll()
+        //{
+        //    var list  = await _resourceGroupLocationAlertCollection.GetAllAsync().ToEnumerableAsync();
+        //    Assert.IsEmpty(list);
+        //}
```
Comment (fengzhou-msft) [inline sdk/securitycenter/Azure.ResourceManager.SecurityCenter/tests/Scenario/ResourceGroupLocationAlertTests.cs:37]: Why commented the test?

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/tests/Scenario/ConnectorSettingTests.cs (hunk 1 above)
Comment (fengzhou-msft) [reply sdk/securitycenter/Azure.ResourceManager.SecurityCenter/tests/Scenario/ConnectorSettingTests.cs:33]: It looks like the [github issue](https://github.com/Azure/azure-rest-api-specs/issues/21260) is already created, please link the issue in the ignore description.

**Review by fengzhou-msft — APPROVED** [review]

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Custom/Models/SecurityAssessmentMetadataData.Serialization.cs [hunk 3]
```diff
@@ -189,6 +187,10 @@ internal static SecurityAssessmentMetadataData DeserializeSecurityAssessmentMeta
                                 property0.ThrowNonNullablePropertyIsNull();
                                 continue;
                             }
+                            if (property0.Value.ValueKind == JsonValueKind.String && string.IsNullOrEmpty(property0.Value.GetString()))
```
Comment (Yao725) [inline sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Custom/Models/SecurityAssessmentMetadataData.Serialization.cs:190]: Considering the type of this property in swagger should be `string`, maybe we can simplify the code here:
```suggestion
                            if (string.IsNullOrEmpty(property0.Value.GetString()))
```

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Custom/SecurityAssessmentMetadataData.cs [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
-// <auto-generated/>
-
 #nullable disable
 
 using System.Collections.Generic;
 using Azure.Core;
 using Azure.ResourceManager.Models;
 using Azure.ResourceManager.SecurityCenter.Models;
 
+[assembly: CodeGenSuppressType("SecurityAssessmentMetadataData")]
```
Comment (Yao725) [inline sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Custom/SecurityAssessmentMetadataData.cs:11]: Why do we need to suppress the generation of all the model? Can we only customize the deserialzation method?

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/tests/Scenario/AlertsSuppressionRuleTests.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        {
+        }
+
+        [SetUp]
+        public void TestSetUp()
+        {
+        }
+
+        [RecordedTest]
+        [Ignore("The SDK doesn't support create a AlertsSuppressionRule")]
```
[the remark this answers is not in the document]
Comment (dvbb) [reply sdk/securitycenter/Azure.ResourceManager.SecurityCenter/tests/Scenario/AlertsSuppressionRuleTests.cs]: I used to think that `AlertsSuppressionRules_Update` meant that creation is not supported. Corresponding tests have been added

**Pull request merged**